### ISSUE TITLE
Clarify purpose, scope, targets, and structure of library

### ISSUE THREAD

**Issue by kmalakoff** [stdlib-js/stdlib#73]
I saw your presentations on math and machine learning on YouTube from a recent Node.js conference and checked this repo out.

This library seems to be doing much more than what I would expect from a standard library in the Node.js ecosystem (eg. small modules with confined scope) and to rely on C++, a module system, etc for the libraries so it seems to be more of a Node.js standard library than a JavaScript standard library (eg. this organization is called stdlib-js rather than stdlib-node).

I was interested in the machine learning functionality and math functionality separately, but it looks like these domains are contained in the same [npm module](https://www.npmjs.com/package/@stdlib/stdlib) and there is a lot of other stuff in this module like plots, datasets, streams, etc.

Do you have plans to break up this into a series of clearly-scoped, independent or inter-dependent modules? In addition to having clear separation in architectural layers, I'm thinking that some of this code is JavaScript-only so it makes sense to distribute as hybrid modules that are browser-friendly (eg. no module system needed).

This looks very interesting and I appreciate these efforts to make Node.js a better platform for science. Great insights and solutions!

**Comment by Planeshifter** [comment]
Thank you for your interest in this library!

`stdlib` is still in an experimental state and under active development, and right now is only usable as a full package. However, the project has been deliberately structured in a way that facilitates publishing individual packages, and in the near future one will be able to install and require both individual functions as well as sub-namespaces like all `math` or `ml` functions. 

While the project uses the Common.JS module system from Node.js, running `stdlib` does not depend on native Node.js add-ons written in C++. Languages like C++ are only required for development, e.g. the generation of test fixtures to test our implementations against reference implementations. While there are some modules in `stdlib` that depend on Node.js such as the facilities to interact with the file system, most modules will work in the Browser after bundling with [browserify](http://browserify.org/). For many functions, polyfills are provided to ensure that the functions work in a variety of environments, including older browsers.

Please let us know if you have more questions.

**Comment by kmalakoff** [comment]
Thank you for the quick response! It sounds like a good direction...

I would recommend as someone looking at the project to quickly assess it (eg. how does it define the scope and domain of a stdlib, is this for node or general JavaScript, etc) that given the focus on C++ and non-JavaScript tooling on the README, that it threw me off. Also, as I watched the videos and long paths to files being used, it looked like it was targeting Node.js only. Perhaps starting the README by focussing on introducing the library (and maybe splitting the development information into a separate file) instead of the focus on development tooling and a adding quick mention that it requires webpack, browserify, etc for the browser would help "Clarify purpose, scope, targets, and structure of library".

Also, the top level structure has quite a few folders and files so it is hard to understand where to dive in to get started and it caused me at least to question the tool chain for the library (and assume it was for a native module)...perhaps simplifying the repository structure to hide some of the complexity (eg. limiting to src, scripts, test folders) and/or breaking down this module using an existing pattern like [lerna](https://lernajs.io/) to signal that it will be published in multiple modules, might help other people be able to assess and use the project...it definitely would put my mind at ease for when I see the C++ tooling and so many topics in one repo.

This looks like a great set of tools!

**Comment by kgryte** [comment]
@kmalakoff We took your suggestion and moved development instructions to a separate file. In their place, we tried to provide more clarity as to project scope and better introduce the library.

Re: lerna. We use a custom build process to decompose the entire library into individual packages. We have yet to take this to production, however, because we are still modifying the project namespace. As soon as we begin publishing individual packages, we commit ourselves to a particular project topology. And given the significance of this commitment, we are holding off until we have a high degree of certainty that the namespace will not change.

Needless to say, the project is still very much in development. :) Thank you for your feedback!
